core/src/main/java/org/opensearch/sql/calcite/CalciteRelNodeVisitor.java

```diff
@@ -1604,53 +1604,64 @@ public RelNode visitStreamWindow(StreamWindow node, CalcitePlanContext context)
           leftWithSeg,
           node,
           groupList,
-          SEQ_COL,
+          ROW_NUMBER_COLUMN_NAME_STREAMSTATS,
           SEGMENT_ID_COL,
-          new String[] {SEQ_COL, RESET_BEFORE_FLAG_COL, RESET_AFTER_FLAG_COL, SEGMENT_ID_COL});
+          new String[] {
+            ROW_NUMBER_COLUMN_NAME_STREAMSTATS,
+            RESET_BEFORE_FLAG_COL,
+            RESET_AFTER_FLAG_COL,
+            SEGMENT_ID_COL
+          });
     }
 
     // CASE: global=true + window>0 + has group
     if (node.isGlobal() && hasWindow && hasGroup) {
       // 1. Add global sequence column for sliding window
       RexNode streamSeq =
-          PlanUtils.makeOver(
-              context,
-              BuiltinFunctionName.ROW_NUMBER,
-              null,
-              List.of(),
-              List.of(),
-              List.of(),
-              WindowFrame.toCurrentRow());
-      context.relBuilder.projectPlus(context.relBuilder.alias(streamSeq, SEQ_COL));
+          context
+              .relBuilder
+              .aggregateCall(SqlStdOperatorTable.ROW_NUMBER)
+              .over()
+              .rowsTo(RexWindowBounds.CURRENT_ROW)
+              .as(ROW_NUMBER_COLUMN_NAME_STREAMSTATS);
+      context.relBuilder.projectPlus(streamSeq);
       RelNode left = context.relBuilder.build();
 
       // 2. Run correlate + aggregate
       return buildStreamWindowJoinPlan(
-          context, left, node, groupList, SEQ_COL, null, new String[] {SEQ_COL});
+          context,
+          left,
+          node,
+          groupList,
+          ROW_NUMBER_COLUMN_NAME_STREAMSTATS,
+          null,
+          new String[] {ROW_NUMBER_COLUMN_NAME_STREAMSTATS});
     }
 
     // Default
-    RexNode streamSeq =
-        PlanUtils.makeOver(
-            context,
-            BuiltinFunctionName.ROW_NUMBER,
-            null,
-            List.of(),
-            List.of(),
-            List.of(),
-            WindowFrame.toCurrentRow());
-    context.relBuilder.projectPlus(context.relBuilder.alias(streamSeq, SEQ_COL));
+    if (hasGroup) {
+      // only build sequence when there is by condition
+      RexNode streamSeq =
+          context
+              .relBuilder
+              .aggregateCall(SqlStdOperatorTable.ROW_NUMBER)
+              .over()
+              .rowsTo(RexWindowBounds.CURRENT_ROW)
+              .as(ROW_NUMBER_COLUMN_NAME_STREAMSTATS);
+      context.relBuilder.projectPlus(streamSeq);
+    }
 
     List<RexNode> overExpressions =
         node.getWindowFunctionList().stream().map(w -> rexVisitor.analyze(w, context)).toList();
     context.relBuilder.projectPlus(overExpressions);
 
     // resort when there is by condition
     if (hasGroup) {
-      context.relBuilder.sort(context.relBuilder.field(SEQ_COL));
+      context.relBuilder.sort(context.relBuilder.field(ROW_NUMBER_COLUMN_NAME_STREAMSTATS));
+      context.relBuilder.projectExcept(
+          context.relBuilder.field(ROW_NUMBER_COLUMN_NAME_STREAMSTATS));
```

Comment:
`context.relBuilder.projectExcept` was out of the `hasGroup` branch, now you move it into the branch. Is it on purpose?